src/USER-BOCS/fix_bocs.cpp

```diff
@@ -200,19 +207,18 @@ FixBocs::FixBocs(LAMMPS *lmp, int narg, char **arg) :
       } else if (strcmp(arg[iarg], "linear_spline") == 0  ) {
         if (iarg+2 > narg) error->all(FLERR,"Illegal fix bocs command. "
                               "Supply a file name after linear_spline.");
-        p_basis_type = 1;
+        p_basis_type = BASIS_LINEAR_SPLINE;
         spline_length = read_F_table( arg[iarg+1], p_basis_type );
         iarg += 2;
       } else if (strcmp(arg[iarg], "cubic_spline") == 0 ) {
         if (iarg+2 > narg) error->all(FLERR,"Illegal fix bocs command. "
                                "Supply a file name after cubic_spline.");
-        p_basis_type = 2;
+        p_basis_type = BASIS_CUBIC_SPLINE;
         spline_length = read_F_table( arg[iarg+1], p_basis_type );
         iarg += 2;
       }  else {
-        char errmsg[256];
-        snprintf(errmsg,256,"CG basis type %s is not recognized\nSupported "
-                 "basis types: analytic linear_spline cubic_spline",arg[iarg]);
+        std::string errmsg = fmt::format("CG basis type {} is not recognized\nSupported "
```

Comment:
This kind of statement doesn't really need a temporary std::string type instance. you can put the call to fmt::format() right  as an argument to error->all().

Reply:
I made another pass through and moved most messages formatting directly into the call to the error method.
I kept formatting to a temp string in a couple of cases where the formatted output was long/complex and, IMO, an embedded call would be hard to read.